README.md

```diff
@@ -28,7 +28,38 @@ source ~/.cargo/env
 ./scripts/qemu.sh
 ```
 
-### Flashing firmware manually is not recommended for the normal user. For the advanced user, there is a script flash.sh which takes the same arguments as build.sh
+## Flashing
+
+**Flashing firmware manually is *not* recommended for the normal user!**
+
+### Internal programmer
+
+Use this method for flashing a system already running System76 Open Firmware.
+
+```
+./scripts/flash.sh <model>
+```
+
+### External programmer
+
+Use this method for first-time flashing or flashing a bricked system.
+
+1. Turn off the computer
+2. Remove the bottom panel
+3. Attach the clip to the SPI ROM chip
+4. Connect to the Raspberry Pi and wait it for boot
+```
+sudo tio -b 115200 /dev/ttyUSB0
+```
+5. From the RPi session, test that the chip is detected.
+```
+cd firmware
+./spipi.sh -r backup.rom
```

Comment:
This is all in a private repository where as this repository is open. Therefore I am not sure this content belongs here.

Reply:
Are you referring to this file specifically? `./scripts/spipi-*.sh` already reference this file (this being a simplified `spipi-dump.sh` to test the SPI clip connection), and it's contents are nothing worth keeping secret.

Reply:
This is good with open sourcing the spipi script and all related code.